lib/discover/global.go

```diff
@@ -208,8 +208,10 @@ func (c *globalClient) sendAnnouncement(timer *time.Timer) {
 	}
 
 	if len(ann.Addresses) == 0 {
-		c.setError(errors.New("nothing to announce"))
-		l.Debugln("Nothing to announce")
+		// There are legitimate cases for not having anything to announce,
+		// yet still using global discovery for lookups. Do not error out
+		// here.
+		c.setError(nil)
 		timer.Reset(announceErrorRetryInterval)
```

Comment:
Shouldn't this be `defaultReannounceInterval`?
I know this PR is already merged (are we very productive today? ;) ), but still seems like the right place to nag about it.

Reply:
I think not, because it is sort of an error that we have no addresses to announce, possibly a temporary one. If so we gain by checking again a bit sooner. Clearly this is subjective, but it was at least intentional and not by mistake I retained that interval. :)